src/NotepadNext/dialogs/MainWindow.cpp

```diff
@@ -1319,7 +1303,15 @@ void MainWindow::bringWindowToForeground()
 
         AttachThreadInput(GetWindowThreadProcessId(GetForegroundWindow(), nullptr), GetCurrentThreadId(), FALSE);
     }
+#else
+    show();
```

Comment:
The linux code path seems to work for windows as well - tried it myself:

Scenario: I open NotepadNext, I open explorer - and while explorer is open, I drag a window into NotepadNext - with the "QtOnlyAPIs(tm)" - the window gets to the front. If I uncomment everything on this function - window still is on background.

Tested on Windows 10 with Qt 6.3.

Question: do you really need all that ifdef... or can you just keep the Linux branch on all platforms?

Reply:
I can try it again on Qt6.3 when I get some free time but I believe the original need for this was in the case of using the right-click context menu to "Open with Notepad Next". Windows does not like it when an application has no interaction (i.e. not getting something dropped on it) and it wants to force its way to the foreground.

Reply:
Tried it again and there is 1 specific scenario that I just can't seem to handle with just Qt calls. If Notepad Next is in the background (e.g. another window is partially in front of it) and the user right-clicks a file and selects "Open with Notepad Next" then it won't change the z order of the application's window. Every other scenario I tested works, even if the application is minimized it will still restore itself and go to the foreground.

I even tried several suggestions here: https://stackoverflow.com/questions/6087887/bring-window-to-front-raise-show-activatewindow-don-t-work

So unfortunately I think this needs to keep the Win32 API calls.